sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs

```diff
@@ -129,12 +130,42 @@ protected ServiceBusSender()
             CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNull(message, nameof(message));
+            await SendAsync(
+                new ServiceBusMessage[] { message },
+                cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        ///   Sends a set of messages to the associated Service Bus entity using a batched approach.
+        ///   If the size of the messages exceed the maximum size of a single batch,
+        ///   an exception will be triggered and the send will fail. In order to ensure that the messages
+        ///   being sent will fit in a batch, use <see cref="SendBatchAsync"/> instead.
+        /// </summary>
+        ///
+        /// <param name="messages">The set of messages to send.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        public virtual async Task SendAsync(
+            IEnumerable<ServiceBusMessage> messages,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(messages, nameof(messages));
             Argument.AssertNotClosed(IsDisposed, nameof(ServiceBusSender));
+            IList<ServiceBusMessage> messageList = messages.ToList();
```

Comment:
@JoshLove-msft @ShivangiReja I think I brought up this discussion in the past so I'm going to bring this up again given that we are reintroducing this API. I can understand that from a broadness of the API perspective the parameter being of type `IEnumerable<>` makes sense. The consequences though are that because the underlying APIs always expose `IList<>` we always need to materialize and copy the enumerable into a list. In what cases would the user not pass in here an already materialized list, I fail to see the use cases because the user always has to create the number of messages upfront so that they can be sent. From that perspective why not exposing `IList` directly or even better `IReadonlyCollection` or something similar and then express that also in the hierarchy of things down the lower layers so that no copying is necessary?

Reply:
The user might pass in an IEnumerable when using `Enumerable.Select` and creating the messages to send in the lambda. I see the fact that we are copying to a list as in implementation detail that shouldn't restrict the API. It is somewhat unfortunate that we have to copy to a list here, as we only do it to include the number of messages being sent in our logs.

Reply:
Is that true given that the virtual APIs also expose IList and AMQP uses list semantics sometimes underneath which means we have to pass IList anyway?

Reply:
The AmqpSender class is internal so users wouldn't be seeing the Send that takes an IList.

Reply:
Uff how could I have forgotten that... Sorry for derailing